src/main/java/com/aws/greengrass/mqttclient/MqttClient.java

```diff
@@ -117,6 +121,9 @@ public void onConnectionInterrupted(int errorCode) {
         @Override
         public void onConnectionResumed(boolean sessionPresent) {
             setMqttOnline(true);
+            if (isSpooling.compareAndSet(false, true)) {
+                executorService.execute(() -> spoolMessages());
```

Comment:
this needs to become more robust now that it isn't scheduled. If `spoolMessages` throws an unchecked exception then it won't start executing again until this callback is triggered.

Reply:
I also don't think that this is sufficient. This callback is only called when the connection is first offline, then online again, so if the device never went offline the spooler would never run.

Reply:
The "onConnectionResumed" will be also run when the connection is firstly connected. https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/main/src/main/java/com/aws/greengrass/mqttclient/AwsIotMqttClient.java#L175

Reply:
> this needs to become more robust now that it isn't scheduled. If `spoolMessages` throws an unchecked exception then it won't start executing again until this callback is triggered.

This is a good callout, it is necessary to make sure the spooler is up and running as expected.